tests/run/cline_in_traceback.srctree

```diff
@@ -0,0 +1,226 @@
+
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_on", ["test_on.pyx"],
+            define_macros=[("CYTHON_CLINE_IN_TRACEBACK", 1)],
+    )))
+    
+######## setup_off.py ########
+
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_off", ["test_off.pyx"],
+            define_macros=[("CYTHON_CLINE_IN_TRACEBACK", 0)],
+    )))
+
+######## setup_runtime.py ########
+
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_runtime", ["test_runtime.pyx"],
+            define_macros=[("CYTHON_CLINE_IN_TRACEBACK_RUNTIME", 1)],
+    )))
+    
+######## setup_default.py ########
+
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_default", ["test_default.pyx"]
+    )))
+    
+######## setup_options_on.py ########
+
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_options_on", ["test_options_on.pyx"]),
+        c_line_in_traceback = True
+    ))
+    
+####### setup_options_off.py #########
+
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_options_off", ["test_options_off.pyx"]),
+        c_line_in_traceback = False
+    ))
+    
+####### setup_options_off_overridden1.py #########
+
+# less important test, but macros should be the ultimate decider
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_options_off_overridden1", ["test_options_off_overridden1.pyx"],
+        define_macros=[("CYTHON_CLINE_IN_TRACEBACK", 1)],
+    ), c_line_in_traceback=False))
+    
+####### setup_options_off_overridden2.py #########
+
+# less important test, but macros should be the ultimate decider
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_options_off_overridden2", ["test_options_off_overridden2.pyx"],
+        define_macros=[("CYTHON_CLINE_IN_TRACEBACK_RUNTIME", 1)],
+    ), c_line_in_traceback = False))
+ 
+####### setup_options_on_overridden.py #########
+
+# less important test, but macros should be the ultimate decider
+from Cython.Build.Dependencies import cythonize
+from setuptools import setup, Extension
+
+setup(
+    ext_modules = cythonize(
+        Extension("test_options_on_overridden", ["test_options_on_overridden.pyx"],
+        define_macros=[("CYTHON_CLINE_IN_TRACEBACK", 0)],
+    ), c_line_in_traceback = True))
+
```

Comment:
I didn't see the case of explicitly contradicting macros covered, so I added it.
```suggestion
####### setup_options_on_overridden.py #########

# less important test, but macros should be the ultimate decider
from Cython.Build.Dependencies import cythonize
from setuptools import setup, Extension

setup(
    ext_modules = cythonize(
        Extension("test_options_on_overridden", ["test_options_on_overridden.pyx"],
        define_macros=[("CYTHON_CLINE_IN_TRACEBACK", 0)],
    ), c_line_in_traceback = True))

####### setup_contradicting_macros.py #########

# less important test, but macros should be the ultimate decider
from Cython.Build.Dependencies import cythonize
from setuptools import setup, Extension

setup(
    ext_modules = cythonize(
        Extension("test_contradicting_macros", ["test_contradicting_macros.pyx"],
        define_macros=[("CYTHON_CLINE_IN_TRACEBACK", 0), ("CYTHON_CLINE_IN_TRACEBACK_RUNTIME", 1)],
    ), c_line_in_traceback = True))

```